src/renderer/components/ContextMenu.tsx

```diff
@@ -1,15 +1,28 @@
 import React, { useState, useEffect, useLayoutEffect, useContext } from 'react'
+import classNames from 'classnames'
 import { useRef } from 'react'
 import { ScreenContext, unwrapContext } from '../contexts'
 
-export type ContextMenuItem = { label: string; action: () => void }
+type ContextMenuItemActionable = { action: () => void; subitems?: never }
+type ContextMenuItemExpandable = {
+  action?: never
```

Comment:
I think this will mean `never | undefined`, better change `?:` to just `:`
same for subitems above.

Reply:
This won't work. I don't remember exact reasoning, but I did try it and from the error output I believe it works like this: since it's impossible to assign any value to a `never` type compiler will never pick `ContextMenuItemActionable` and it then complains about either `subitems` not specified in cases where it's absent or that it cannot assign a function to `never` in case where `action` is specified. It's a common trick to use optional never for this.